tests/deployment/sagemaker/test_sagemaker.py

```diff
@@ -1,33 +1,419 @@
+def sagemaker_client():
+    return boto3.client('sagemaker', 'us-west-2')
 
-def test_sagemaker_handle_client_errors():
-    client = boto3.client('sagemaker', 'us-west-2')
-    stubber = Stubber(client)
+
+def test_sagemaker_handle_client_errors(sagemaker_client):
+    sagemaker_client = boto3.client('sagemaker', 'us-west-2')
+    stubber = Stubber(sagemaker_client)
 
     stubber.add_client_error(
         method='create_endpoint', service_error_code='ValidationException'
     )
     stubber.activate()
     result = None
-    try:
-        client.create_endpoint(EndpointName='Test', EndpointConfigName='test-config')
-    except ClientError as e:
-        result = _parse_aws_client_exception_or_raise(e)
+
+    with pytest.raises(ClientError) as error:
+        sagemaker_client.create_endpoint(
+            EndpointName='Test', EndpointConfigName='test-config'
+        )
+    result = _parse_aws_client_exception_or_raise(error.value)
 
     assert result.status_code == Status.NOT_FOUND
 
     stubber.add_client_error('describe_endpoint', 'InvalidSignatureException')
     stubber.activate()
     result = None
-    try:
-        client.describe_endpoint(EndpointName='Test')
-    except ClientError as e:
-        result = _parse_aws_client_exception_or_raise(e)
+    with pytest.raises(ClientError) as e:
+        sagemaker_client.describe_endpoint(EndpointName='Test')
+    result = _parse_aws_client_exception_or_raise(e.value)
     assert result.status_code == Status.UNAUTHENTICATED
+
+
+def test_cleanup_sagemaker_model(sagemaker_client):
+    sagemaker_client = boto3.client('sagemaker', 'us-west-2')
+    stubber = Stubber(sagemaker_client)
+    stubber.add_client_error(
+        method='delete_model', service_error_code='ValidationException'
+    )
+    stubber.activate()
+
+    error_status = _cleanup_sagemaker_model(
+        sagemaker_client, 'test_name', 'test_version'
+    )
+    assert error_status.status_code == Status.NOT_FOUND
+
+    stubber.add_client_error(
+        method='delete_model', service_error_code='InvalidSignatureException'
+    )
+    error_status = _cleanup_sagemaker_model(
+        sagemaker_client, 'test_name', 'test_version'
+    )
+    assert error_status.status_code == Status.UNAUTHENTICATED
+
+    stubber.add_client_error(
+        method='delete_model',
+        service_error_code='RandomError',
+        service_message='random',
+    )
+    with pytest.raises(ClientError) as error:
+        _cleanup_sagemaker_model(sagemaker_client, 'test_name', 'test_version')
+    assert error.value.operation_name == 'DeleteModel'
+    assert error.value.response['Error']['Code'] == 'RandomError'
+
+
+def test_cleanup_sagemaker_endpoint_config(sagemaker_client):
+    stubber = Stubber(sagemaker_client)
+    stubber.add_client_error(
+        method='delete_endpoint_config', service_error_code='ValidationException'
+    )
+    stubber.activate()
+
+    error_status = _cleanup_sagemaker_endpoint_config(
```

Comment:
```_cleanup_sagemaker_endpoint_config``` really should not be returning an error status, will definitely need to do some refactoring there, could you just remove these two tests `test_cleanup_sagemaker_model` and `test_cleanup_sagemaker_endpoint_config`  for now? they are just testing one line logic anyway